jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/HiveUrlUtils.java

```diff
@@ -0,0 +1,163 @@
+ * limitations under the License.
+ */
+package com.google.cloud.dataproc.jdbc;
+
+import static com.google.cloud.dataproc.jdbc.DataprocDriver.DATAPROC_JDBC_HIVE_URL_SCHEMA;
+
+import java.io.IOException;
+import java.net.URI;
+
+/** Helper methods to establish Connection for DataprocDriver. */
+public class HiveUrlUtils {
+    public static final String HIVE_DEFAULT_DATABASE = "";
+    public static final int HIVE_DEFAULT_PORT = 10001;
+    public static final String URL_JDBC_PREFIX = "jdbc:";
+
+    /**
+     * Parses client url and extracts Hive connection parameters.
+     *
+     * @param url verified url passed in by client, already of valid format possible formats:
+     *     jdbc:dataproc://hive/<dbName>;clusterName=<>;projectId=<>;region=<>;port=<>
+     *     jdbc:dataproc://hive/<dbName>;clusterName=<>;projectId=<>;region=<>;clusterPoolLabel=foo=bar:env=staging;
+     * @return HiveJdbcConnectionParam object with fields correctly set
+     * @throws IOException
+     * @throws InvalidURLException
+     */
+    public static HiveJdbcConnectionOptions parseHiveUrl(String url) throws InvalidURLException {
+        if (url.equals(DATAPROC_JDBC_HIVE_URL_SCHEMA)) {
+            throw new InvalidURLException(
+                    String.format("Error URL: %s \n" + "Please provide cluster information.", url));
+        }
+
+        HiveJdbcConnectionOptions.Builder paramBuilder = HiveJdbcConnectionOptions.builder();
+
+        String urlStripped = url.substring(URL_JDBC_PREFIX.length());
+
+        URI jdbcURI = URI.create(urlStripped);
+
+        // getPath will return: /<dbName>;<sessionConfs>
+        String dbAndSessionConfs = jdbcURI.getPath().substring(1);
+
+        // no dbName or no cluster info
+        if (dbAndSessionConfs.indexOf(";") == -1) {
+            throw new InvalidURLException(
+                    String.format(
+                            "Error: %s \n"
+                                    + "Please provide dbName or empty if no dbName specified.\n"
+                                    + "Please provide cluster information.",
+                            dbAndSessionConfs));
+        }
+
+        String dbName = dbAndSessionConfs.substring(0, dbAndSessionConfs.indexOf(";"));
+        paramBuilder.setDbName(dbName.isEmpty() ? HIVE_DEFAULT_DATABASE : dbName);
+
+        // getQuery() will return the info after '?'
+        String hiveConfs = jdbcURI.getQuery();
+        // getFragmeng() will return the info after '#
+        String hiveVars = jdbcURI.getFragment();
+
+        // Params that do not need parsing
+        paramBuilder.setHiveConfs(hiveConfs);
+        paramBuilder.setHiveVars(hiveVars);
+
+        // Parse sessionConfigs to get cluster info
+        String sessionConfigs = dbAndSessionConfs.substring(dbName.length() + 1);
+        String[] params = sessionConfigs.split(";");
+
+        StringBuilder otherSessionConfsBuilder = new StringBuilder();
```

Comment:
Nit: this might be a bit clearer as a List. And then `toJdbcUrl()` can just join the list on `;` and append: https://stackoverflow.com/questions/1751844/java-convert-liststring-to-a-string